src/Cache/_ScsDataClient.php

```diff
@@ -224,4 +249,119 @@ public function listPushBack(
+            $call = $this->grpcManager->client->ListPopBack(
+                $listPopBackRequest, ["cache"=>[$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]
+            );
+            $response = $this->processCall($call);
+        } catch (SdkError $e) {
+            return new CacheListPopBackResponseError($e);
+        } catch (\Exception $e) {
+            return new CacheListPopBackResponseError(new UnknownError($e->getMessage()));
+        }
+        if (!$response->hasFound())
+        {
+            return new CacheListPopBackResponseMiss();
+        }
+        return new CacheListPopBackResponseHit($response);
+    }
+
+    public function listRemoveValue(string $cacheName, string $listName, string $value) : CacheListRemoveValueResponse
+    {
+        try {
+            validateCacheName($cacheName);
+            validateListName($listName);
+            $listRemoveValueRequest = new _ListRemoveRequest();
+            $listRemoveValueRequest->setListName($listName);
+            $listRemoveValueRequest->setAllElementsWithValue($value);
+            $call = $this->grpcManager->client->ListRemove(
+                $listRemoveValueRequest, ["cache"=>[$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]
+            );
+            $this->processCall($call);
+        } catch (SdkError $e) {
+            return new CacheListRemoveValueResponseError($e);
+        } catch (\Exception $e) {
+            return new CacheListRemoveValueResponseError(new UnknownError($e->getMessage()));
+        }
+        return new CacheListRemoveValueResponseSuccess();
+    }
+
+    public function listLength(string $cacheName, string $listName) : CacheListLengthResponse
+    {
+        try {
+            validateCacheName($cacheName);
+            validateListName($listName);
+            $listLengthRequest = new _ListLengthRequest();
+            $listLengthRequest->setListName($listName);
+            $call = $this->grpcManager->client->ListLength(
+                $listLengthRequest, ["cache"=>[$cacheName]], ["timeout" => $this->deadline_seconds * self::$TIMEOUT_MULTIPLIER]
+            );
+            $response = $this->processCall($call);
+        } catch (SdkError $e) {
+            return new CacheListLengthResponseError($e);
+        } catch (\Exception $e) {
+            return new CacheListLengthResponseError(new UnknownError($e->getMessage()));
+        }
+        return new CacheListLengthResponseSuccess($response);
+    }
+
+    public function listErase(string $cacheName, string $listName, ?int $beginIndex=null, ?int $count=null) : CacheListEraseResponse
+    {
+        try {
+            validateCacheName($cacheName);
+            validateListName($listName);
+            validateRange($beginIndex, $count);
+            $listEraseRequest = new _ListEraseRequest();
+            $listEraseRequest->setListName($listName);
+            if (!is_null($beginIndex) && !is_null($count)) {
```

Comment:
Do we need this if we are already checking with `validateRange`?

Reply:
Oh I see. Those are optional, and if they're null, delete all the items.

Reply:
Yes. This is not a validation of the data. If these values are both supplied we need to erase a range of items and if neither value is supplied we need to erase the entire list. The validation makes sure that one or the other of those is true.